Libraries/DirectMLX.h

```diff
@@ -4253,6 +4294,19 @@ namespace dml
             return { NodeType::Reinterpret, index };
         }
 
+        inline NodeID GraphBuilder::CreateConstantNode(Span<const Byte> data)
+        {
+            uint32_t index = static_cast<uint32_t>(m_constantNodes.size());
+            m_constantNodes.push_back(ConstantNode{ data });
+
+            if (!m_name.empty())
+            {
+                m_constantNodes.back().name = m_name;
```

Comment:
>m_name [](http://example.com/codeflow?start=46&length=6)

Wouldn't this variable represent the graph name because it is member variable of the GraphBuilder class?
I think we should also accept the constant node name from the user as part of `ConstantData` method or generate a unique name if not provided.

Reply:
It's not the name of the graph, but rather the "current name scope" of the graph. It's not easy to name nodes in a graph with DMLX since they aren't always explicit. For example, say you have a dmlx graph to do a quadratic expression:

```
dml::Expression y = a * x * x + b * x + c;
```

How do you name the multiply/add nodes in the graph above? They're implicitly created through operator overloading, so you don't have a reference to the dml::Expression. The GraphBuilder has a name scope so you can do something like this:

```
graph->PushName("term1");
auto y = a * x * x;
graph->PopName();

graph->PushName("term2");
y += b * x;
graph->PopName();

graph->PushName("term3");
y += c;
graph->PopName();
```

Any nodes created while there's a name scope active inherit the current name, so the first two multiply nodes would get the name "term1".

Reply:
Interesting case, given the constant nodes have explicit names.

```
struct DML_CONSTANT_DATA_GRAPH_NODE_DESC 
{ 
    _Field_size_bytes_(DataSize) const void* Data;
    SIZE_T DataSize;
    _Field_z_ _Maybenull_ const char* Name; 
}; 
```

If a constant node *has* a name, then presumably it has priority over any default pushed name?